uaclient/config.py

```diff
@@ -121,6 +122,14 @@ def __init__(
             self.cfg_path = get_config_path()
             self.cfg, self.invalid_keys = parse_config(self.cfg_path)
 
+        if user_config:
+            self.user_config = user_config
+        else:
+            self.user_config = (
+                state_files.user_config_file.read()
+                or state_files.UserConfigData()
```

Comment:
Should we support this scenario where we the config json file doesn't exist ? We will create user config with all `None` values which can cause errors in down the line or even worse, cause an unexpected result  (i.e. the user relying on a proxy for example)

Reply:
Most of the getters on UAConfig for these fields return `Optional` already so the type signature won't change and in theory mypy has checked that we don't assume they're not None anywhere.
For the rest, we'll implement default values in UAConfig.
Do you think this is sufficient?

Reply:
My only concern here is the corner case where the user had a valid config that was corrupted somehow and know we will be running commands with default values. For example, if the user has a proxy configured in the old configuration.

I know this should be really rare, but I think we should alert the user somehow that this is happening